kernel/aster-nix/src/syscall/getdents64.rs

```diff
@@ -14,6 +14,31 @@ use crate::{
     util::write_bytes_to_user,
 };
 
+pub fn sys_getdents(fd: FileDesc, buf_addr: Vaddr, buf_len: usize) -> Result<SyscallReturn> {
+    log_syscall_entry!(SYS_GETDENTS);
+    debug!(
+        "fd = {}, buf_addr = 0x{:x}, buf_len = 0x{:x}",
+        fd, buf_addr, buf_len
+    );
+    let file = {
+        let current = current!();
+        let file_table = current.file_table().lock();
+        file_table.get_file(fd)?.clone()
+    };
+    let inode_handle = file
+        .downcast_ref::<InodeHandle>()
+        .ok_or(Error::with_message(Errno::EBADF, "not inode"))?;
+    if inode_handle.dentry().type_() != InodeType::Dir {
```

Comment:
This might rely on the #758 .

Reply:
You are right. I will mark it as a draft.

Reply:
After a systematic renaming, you might want to use dentry() again after the PR is merged.

PR #758 takes a better naming convention, which uses `Dentry` to replace `DentryMnt` and uses `Dentry_` to replace the original `Dentry`, so that users can keep their function names and it introduces less code modification.

Reply:
> After a systematic renaming, you might want to use `dentry()` again after the PR is merged.
> 
> PR #758 takes a better naming convention, which uses `Dentry` to replace `DentryMnt` and uses `Dentry_` to replace the original `Dentry`, so that users can keep their function names and it introduces less code modification.

Now the PR has been merged.